### ISSUE TITLE
Create Documentation for Project Structure

### ISSUE THREAD

**Issue by evgongora** [Vagabonds-Labs/cofiblocks#90]
## Objective:

Develop a comprehensive guide to the project structure to help new developers onboard quickly and effectively.

## Steps:

1. Overview of the Project:

Provide a high-level description of the project's purpose and architecture.
Explain how the project is organized into workspaces (e.g., apps, packages).

2. Detailed Breakdown of Key Directories:

apps/:
Describe the main web application and its components.
Include information about pages, components, and hooks.
packages/:
Explain shared utilities, styles, and any other reusable modules.
config/:
Detail configurations (e.g., ESLint, Biome, TailwindCSS).

3. Scripts and Commands:

List and explain key scripts like build, dev, and db:migrate.
Provide examples of how to use each script during development.

Show how modules, components, and services interact.
Include a diagram or chart if possible.

4. Development Guidelines:

Emphasize coding conventions and standards.
Link to the Biome formatting and linting rules.

5. Testing Directory and Standards:

Explain how to write and organize tests (unit, integration, E2E).
Provide examples of existing tests and how to run them.

6. Include Visual Aids:

Add file structure snippets or flow diagrams for clarity.
Use examples to illustrate concepts.

**Comment by Belloabraham121** [comment]
Hey! I'm Iteooluwakisi.
Member of Web3bridge. I have contributed before to other projects like:
• CoinSafe 
• Scaffold-ETH
• BuidlGuidl
Can I work on this issue?
Plan to Solve the Issue:
Study the codebase to understand the current structure and functionality.
Understand the problem that needs to be solved by thoroughly reading the issue details.
Leverage all provided resources within the issue to get a better grasp of the task at hand.
If I encounter roadblocks, I will ask questions ASAP in the relevant group/chat to move forward effectively.
I will open a PR and wait for your feedback.

**Comment by martinvibes** [comment]
Can I take this issue?

**Comment by SYLVIANNORUKA** [comment]
Let me handle this issue! I am a Front-end and smart contract developer. I am proficient in tailwind css, Next Js, Typescript, JavaScript, . I would love to contribute to this project.

I’ll create a guide to help new developers onboard by explaining the project’s structure. This will include an overview of the project’s purpose, key directories like `apps/`, `packages/`, and `config/`, and details on important scripts. I’ll also show how components and services interact, emphasize coding conventions, and explain the testing structure. Finally, I’ll add visual aids, such as file structure snippets and diagrams, for clarity.

**Comment by 0xdevcollins** [comment]
Can I be assigned this issue?

My Background
I am a Full Stack Blockchain Developer with proficiency in Solidity/Rust, Next.js, TypeScript, React, and Node.js. I've demonstrated my ability to adjust to a variety of requirements, perform well under pressure, and reliably produce user-centric blockchain solutions with 58 significant contributions spread over 17 OnlyDust projects.

**Comment by BrunoAmbricca** [comment]
I would like to work on this issue

**Comment by EmmanuelAR** [comment]
Hello team, Im Emmanuel Aguero from Costa Rica and dojo coding.
I will like to work on this, i will follow all the documentation instructions.

Let me know!

**Comment by mimiprosper** [comment]
I would like to take this task. Am a web 2 frontend developer, a technical writer, a solidity smart contract developer/auditor and a cairo smart contract developer. Am a graduate of Starknet Cairo Bootcamp. I have successfully done tasks assigned to me in OnlyDust. 
If assigned this i would complete it within 2 days. 

Here are some articles i have written: https://medium.com/@emma.onyedika.okeke

**Comment by respp** [comment]
Hello! I am part of the Dojo community and am excited about the opportunity to make my first contributions to OnlyDust. 
I would like to develop the documentation since, with my background as a teacher, I can explain in a simple way what the project requires, following all the steps provided by evgongora.
Just let me know when I should start! 🚀

**Comment by Amarjeet325** [comment]
I would like to work on this issue

**Comment by vestor-dev** [comment]
hey sir i would love to work on this issue 
i'm a blockchain dev and i would love to handle this issue 
assign me and i'll get to work

**Comment by Benjtalkshow** [comment]
I am a Full Stack Developer with a strong background in blockchain and extensive experience in Next.js/React, TypeScript, and Rust. I’ve made over 49 contributions to over 17 projects in the OnlyDust ecosystem, solving complex issues and delivering efficient, scalable solutions.

I can handle this task.
ETA is 3days.

**Comment by Michaelkingsdev** [comment]
Could I take a shot at this?

**Comment by mimisavage** [comment]
I’d like to work on this.

**Comment by mimisavage** [comment]
Would love to tackle this!

**Comment by TheSQLguy** [comment]
I am new to open source.
So I will create a full guide describing the project structure, purpose, and organization around the key directories like apps/, packages/, and config/. I will specify key scripts and explain development guidelines and testing standards with examples for clarity. Finally, I will create visual aids such as file structure snippets and flow diagrams to help onboard new developers.

**Comment by olisaagbafor** [comment]
Could I take over this issue?

**Comment by DuendexCR** [comment]
Could I try solving this?

**Comment by kayceeDev** [comment]
I’m interested in this one.
I have experience in writing technical documentation, typescript and using cairo.

I will tackle this issue by investigating the codebase and understanding the project.
I will also develop an intuitive guide to help onboard new developers quickly.

Please assign me.

**Comment by Dorcas18** [comment]
Could I take on this issue?

**Comment by 3th-Enjay** [comment]
May I handle this issue?

**Comment by Jayrodri088** [comment]
Hi, I'm John Asogwa, and I’d like to work on this issue.
I have over three years of experience in TypeScript, React, and Next.js, as well as expertise in documenting complex systems for developers. I’m excited about the vision of Cofiblocks to empower small coffee farmers and create a sustainable future for coffee trade, and I’d love to contribute to making this project more accessible for new developers.

I estimate this task will take 2 business days to complete.

Steps to Resolve
1. High-Level Overview of the Project
Start by analyzing the Cofiblocks GitHub repository to understand its purpose, structure, and key components.
Provide a clear explanation of the project's mission and how its architecture supports the goals of cutting out intermediaries and modernizing the coffee trade using StarkNet.
Document how the project is divided into workspaces (e.g., apps/ and packages/) to improve clarity for new developers.
2. Detailed Breakdown of Key Directories
apps/:

Describe the main web application and its key components, such as:
Pages: Their role in routing and rendering content.
Components: Reusable building blocks of the UI.
Hooks: State and logic management utilities.
Provide examples of how each part interacts with the broader application.
packages/:

Detail shared utilities, such as styles, reusable modules, or any helper functions.
Include practical examples, such as a utility function being used in the web app.
config/:

Explain configurations for tools like ESLint, Biome, and TailwindCSS.
Show how these configurations enforce coding standards and streamline development.
3. Scripts and Commands
Document essential scripts, such as:
pnpm build: Builds the project.
pnpm dev: Starts the development server.
pnpm db:migrate: Applies database migrations (if applicable).
Provide usage examples and explain scenarios where these scripts would be used.
4. Module and Component Interactions
Illustrate how different modules, components, and services interact to build the application.
Include a visual flow diagram or chart to represent the relationships between directories, components, and services.
5. Development Guidelines
Highlight coding conventions and standards.
Link to Biome formatting and linting rules, emphasizing their importance in maintaining code quality.
Provide examples of common patterns used in the codebase (e.g., how to write hooks or create reusable components).
6. Testing Directory and Standards
Document how tests are structured and written, including:
Unit tests for components or utilities.
Integration tests for modules or APIs.
End-to-End (E2E) tests for user flows.
Provide an example test case and show how to run the test suite using commands like:
bash
Copy code
pnpm test
7. Add Visual Aids
Include file structure snippets to clarify how directories are organized.
Add diagrams to illustrate workflows, such as how data flows between apps/ and packages/.
Use examples throughout the README to demonstrate concepts.
Deliverables
Once complete, I will submit:

A detailed README with all the above sections.
Visual diagrams embedded in the README for clarity.
A pull request with a clear description of the changes and links to resources for further reading.
This guide will make onboarding seamless for new developers and help them contribute effectively to Cofiblocks’ mission of building a more equitable and sustainable coffee industry.

Let me know if you’d like me to get started on this! 😊

**Comment by evgongora** [comment]
@Jayrodri088 Any update on this?

**Comment by Jayrodri088** [comment]
Yes, I have been able to create markdown files detailing the sections of the project, I would have a PR ready today

**Comment by evgongora** [comment]
> Yes, I have been able to create markdown files detailing the sections of the project, I would have a PR ready today

Any update?

**Comment by Jayrodri088** [comment]
I've been trying to commit my changes but I cannot due to errors in the contracts folder, on the predeployedContarcats.ts file

**Comment by brolag** [comment]
Hi @Jayrodri088, can you updating your branch and pushing your changes again?

**Comment by Jayrodri088** [comment]
Alright

> n
